pkgs/build-support/rust/default.nix

```diff
@@ -61,14 +68,19 @@ in stdenv.mkDerivation (args // {
     ${setupVendorDir}
 
     mkdir .cargo
-    cat >.cargo/config <<-EOF
-      [source.crates-io]
-      registry = 'https://github.com/rust-lang/crates.io-index'
-      replace-with = 'vendored-sources'
-
-      [source.vendored-sources]
-      directory = '$(pwd)/$cargoDepsCopy'
-    EOF
+  '' + (if useRealVendorConfig then ''
+      sed "s|directory = \".*\"|directory = \"$(pwd)/$cargoDepsCopy\"|g" \
+        "$(pwd)/$cargoDepsCopy/.cargo/config" > .cargo/config
```

Comment:
The fact, that there are so many prior issues and PRs about this is annoying. I really should've done my research, sorry for that.

Also, the fact that the output of cargo-vendor can change really did not come to my mind. As you say it, it is clear to me that my PR wouldn't be the good solution I was hoping for, but more of a hack.

I wonder if it would be better to close this PR and open another one, as your proposal really changes the direction where this is going?

Reply:
I think what could work instead is, having a script with a toml parser (e.x.: python), that reads the cargo config and produces cargo config for the fixed-output derivation. That way we can control the output.